### DISCUSSION TITLE
How to use Table Aliases?

### DISCUSSION THREAD

**Discussion by richb-hanover** [PRQL/prql#1252] [General]
I would like to use (I think they're called) Table Aliases to simplify code. That is, I would like to use:

```
from l=TableA
join r=TableB [==Col1]
select [
  l.Col1,
  r.Col1,
]
filter [l.Col1 != r.Col1]
```

* If I use this form, the Playground gives an error: `Unknown name l.Col1` in the filter expression.

* If I move the filter between the `join` and `select`, I get a long `Cannot lower to IR expr:...` message.

What am I missing? Thanks.

**Comment by max-sixty** [comment]
Good find, this looks like quite a bad bug. Moving to an issue: https://github.com/prql/prql/issues/1253
